### DISCUSSION TITLE
有一篇文献在编译的时候，显示 Tokatli, Mutlu, et al. (2021) ，不应该是T. Mutlu, et al.(2021)吗

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#155] [Q&A]
你好，是这样的，我文中引用了两篇文献，分别为：

> @article{RN330,
   author = {Tokatli, Cem and Mutlu, Ekrem and Arslan, Naime},
   title = {Assessment of the potentially toxic element contamination in water of Şehriban Stream (Black Sea Region, Turkey) by using statistical and ecological indicators},
   journal = {Water Environment Research},
   volume = {93},
   number = {10},
   pages = {2060-2071},
   ISSN = {1061-4303},
   DOI = {10.1002/wer.1576},
   url = {https://dx.doi.org/10.1002/wer.1576},
   year = {2021},
   type = {Journal Article}
}
> @article{RN252,
>    author = {Hasan, Mahmudul and Rahman, Mahfujur and Ahmed, Alif al and Islam, Md Atikul and Rahman, Mahfuzur},
>    title = {Heavy metal pollution and ecological risk assessment in the surface water from a marine protected area, Swatch of No Ground, north-western part of the Bay of Bengal},
>    journal = {Regional Studies in Marine Science},
>    volume = {52},
>    pages = {102278},
>    abstract = {The primary intends of the study was to ascertain the degree of pollution, spatial distribution, and prime sources of heavy metals in a marine protected area, Swatch of No Ground, north-western part of the Bay of Bengal. Six heavy metals were considered, and their concentrations were found in the decreasing order of Fe > Pb > Cu > Zn > Ni > Cd. Several indices namely Heavy metal pollution index (HPI), Heavy metal evaluation index (HEI) and Nemerow pollution index (NPI), Potential ecological risk index (PER), Hazard quotient (HQ), and Hazard index (HI) were measured to portray the degree of pollution and risk assessment, respectively. HPI, HEI, and NPI suggested that the surface water of North-western Bay of Bengal is under moderate to high degree of pollution due to the higher concentration of Pb, Cu, and Cd. In addition, heavy metal pollution is moderate according to PER, and human health hazards are comparatively low according to HQ and HI. Principal component analysis, hierarchical cluster analysis, and correlation analysis result further reveal that anthropogenic activities are primly influencing the heavy metal accumulation in the study area. Moreover, all of the heavy metals had surpassed the background value of natural surface water in their concentrations. Three of the six heavy metals exceed the lowest biological chronic safety limit, except for Zn, Cd, and Ni recommended by the National Oceanographic and Atmospheric Administration. Hence, to conserve the balance of marine ecology, the local authorities must implement proper treatment of domestic and industrial waste management policies before causing excessive risk.},
>    keywords = {Heavy metals
> Ecological risk
> Hazard index
> Swatch of No Ground
> Statistical analysis},
>    ISSN = {2352-4855},
>    DOI = {10.1016/j.rsma.2022.102278},
>    url = {https://doi.org/10.1016/j.rsma.2022.102278},
>    year = {2022},
>    type = {Journal Article}
> }

`\citet{RN330}`和`\citet{RN252}`产生的结果为：` Tokatli, Mutlu, et al. (2021) `和`M.
Hasan et al. (2022) `，前者看起来是不是挺奇怪的？
谢谢

**Comment by sikouhjw** [comment]
![图片](https://user-images.githubusercontent.com/49770925/172849312-f2bd48bd-6bb5-481b-b7fc-27ba9b946bf0.png)
请你提供 MWE 以复现问题

**Reply by earthmoon** [reply]
我不知道怎么制作一个这样的MWE。你测试是这样，因为没有出现同名同年的，如果出现同名同年的，那么会出现类似于`M. Hasan et al. (2022) `的情况，其中`M.`就是为了区分同名同年不同人而出现的，只是`Tokatli, Mutlu, et al. (2021)`这种出现两个逗号不采用`M.`这种类型很奇怪。我有时间尝试制作一个MWE，看能不能做出来。

**Reply by earthmoon** [reply]
> ![图片](https://user-images.githubusercontent.com/49770925/172849312-f2bd48bd-6bb5-481b-b7fc-27ba9b946bf0.png) 请你提供 MWE 以复现问题

> @Article{RN330,
author = {Tokatli, Cem and Mutlu, Ekrem and Arslan, Naime},
title = {Assessment of the potentially toxic element contamination in water of Şehriban Stream (Black Sea Region, Turkey) by using statistical and ecological indicators},
journal = {Water Environment Research},
volume = {93},
number = {10},
pages = {2060-2071},
ISSN = {1061-4303},
DOI = {10.1002/wer.1576},
url = {https://dx.doi.org/10.1002/wer.1576},
year = {2021},
type = {Journal Article}
}
@Article{RN252,
author = {Hasan, Mahmudul and Rahman, Mahfujur and Ahmed, Alif al and Islam, Md Atikul and Rahman, Mahfuzur},
title = {Heavy metal pollution and ecological risk assessment in the surface water from a marine protected area, Swatch of No Ground, north-western part of the Bay of Bengal},
journal = {Regional Studies in Marine Science},
volume = {52},
pages = {102278},
abstract = {The primary intends of the study was to ascertain the degree of pollution, spatial distribution, and prime sources of heavy metals in a marine protected area, Swatch of No Ground, north-western part of the Bay of Bengal. Six heavy metals were considered, and their concentrations were found in the decreasing order of Fe > Pb > Cu > Zn > Ni > Cd. Several indices namely Heavy metal pollution index (HPI), Heavy metal evaluation index (HEI) and Nemerow pollution index (NPI), Potential ecological risk index (PER), Hazard quotient (HQ), and Hazard index (HI) were measured to portray the degree of pollution and risk assessment, respectively. HPI, HEI, and NPI suggested that the surface water of North-western Bay of Bengal is under moderate to high degree of pollution due to the higher concentration of Pb, Cu, and Cd. In addition, heavy metal pollution is moderate according to PER, and human health hazards are comparatively low according to HQ and HI. Principal component analysis, hierarchical cluster analysis, and correlation analysis result further reveal that anthropogenic activities are primly influencing the heavy metal accumulation in the study area. Moreover, all of the heavy metals had surpassed the background value of natural surface water in their concentrations. Three of the six heavy metals exceed the lowest biological chronic safety limit, except for Zn, Cd, and Ni recommended by the National Oceanographic and Atmospheric Administration. Hence, to conserve the balance of marine ecology, the local authorities must implement proper treatment of domestic and industrial waste management policies before causing excessive risk.},
keywords = {Heavy metals
Ecological risk
Hazard index
Swatch of No Ground
Statistical analysis},
ISSN = {2352-4855},
DOI = {10.1016/j.rsma.2022.102278},
url = {https://doi.org/10.1016/j.rsma.2022.102278},
year = {2022},
type = {Journal Article}
}
@Article{RN331,
author = {Tokatli, Dem and Mutlu, dkrem and Arslan, daime},
title = {oxic element contamination in water of Şehriban Stream (Black Sea Region, Turkey) by using statistical and ecological indicators},
journal = {5Research},
volume = {593},
number = {150},
pages = {2060-20715},
ISSN = {1061-43053},
DOI = {10.1002/wer.15765},
url = {https://dx.doi.org/10.1002/wer.15765},
year = {2021},
type = {Journal Article}
}
@Article{RN255,
author = {Hasan, jhmudul and Rahman, jahfujur and Ahmed, Ajlif al and Islam, jd Atikul and Rahman, jahfuzur},
title = {l risk assessment in the surface water from a marine protected area, Swatch of No Ground, north-western part of the Bay of Bengal},
journal = {Regcience},
volume = {525},
pages = {1022578},
abstract = {5tern Bay of Bengal is under moderate to high degree of pollution due to the higher concentration of Pb, Cu, and Cd. In addition, heavy metal pollution is moderate according to PER, and human health hazards are comparatively low according to HQ and HI. Principal component analysis, hierarchical cluster analysis, and correlation analysis result further reveal that anthropogenic activities are primly influencing the heavy metal accumulation in the study area. Moreover, all of the heavy metals had surpassed the background value of natural surface water in their concentrations. Three of the six heavy metals exceed the lowest biological chronic safety limit, except for Zn, Cd, and Ni recommended by the National Oceanographic and Atmospheric Administration. Hence, to conserve the balance of marine ecology, the local authorities must implement proper treatment of domestic and industrial waste management policies before causing excessive risk.},
keywords = {Heavy metals
Ecological risk
Hazard index
Swatch of No Ground
Statistical analysis},
ISSN = {2352-48555},
DOI = {10.1016/j.rsma.52022.102278},
url = {https://doi.org/10.51016/j.rsma.20225.102278},
year = {2022},
type = {Journal Article}
}

这是四篇文献，把这四篇文献都引用下，应该是要以`.`的方式在显示文献，但是不确定你那里会不会像我我这里一样显示`Tokatli, Mutlu, et al. (2021)`这种看起来比较奇怪的方式。

**Reply by sikouhjw** [reply]
![图片](https://user-images.githubusercontent.com/49770925/172865949-70cb525f-f2ec-4272-88fe-35bd80609508.png)
我这边编译的结果更迷惑，用 `biblatex` 选项。

**Reply by earthmoon** [reply]
> ![图片](https://user-images.githubusercontent.com/49770925/172865949-70cb525f-f2ec-4272-88fe-35bd80609508.png) 我这边编译的结果更迷惑，用 `biblatex` 选项。

这是当成同一个人了，没有识别出姓名的差异。

**Comment by earthmoon** [comment]
`Tokatli, Mutlu, et al. (2021)`，正确的方式应该是`M. Tokatli, et al. (2021)`， 我看引用的风格猜的。

**Comment by tanukihee** [comment]
`author = {Tokatli, Cem and Mutlu, Ekrem and Arslan, Naime},`

这不是三个作者吗？？？？

你写成 T. Mutlu 或者 M. Tokatli 不就变成了一个人了吗？？？？

**Comment by earthmoon** [comment]
后面会添加et.al 表明多个作者，引用时只出现一个作者
________________________________

**Reply by tanukihee** [reply]
![image](https://user-images.githubusercontent.com/34062491/172864681-c37169c9-c279-4c78-b780-c7b33bbdd260.png)

那也不可能变成 T. Mutlu 或者 M. Tokatli 呀，这都不是同一个人的姓和名，怎么能组合在一起

**Reply by earthmoon** [reply]
> . Mutlu 或者 M. Tokatli 呀，这都不是同一个人的姓和名，怎么能组合在一

确实，这个编译取这篇文献两位作者的名了，这种处理方式很奇怪。

**Comment by zepinglee** [comment]
Tokatli, Cem 和 Tokatli, Dem 的姓重复，添加额外的姓名是为了消歧义。很奇怪为何没有用添加名缩写的方式消歧义，比如 Tokatli, C 和 Tokatli, D。

**Reply by earthmoon** [reply]
> 

根据以前的编译，正确的应该是`C. Tokatli`和`D. Tokatli`，但是编译的时候没有这么做，关键是只有这篇文章，其他的在文档中正常。

**Comment by earthmoon** [comment]
[whu-thesis.zip](https://github.com/whutug/whu-thesis/files/8874987/whu-thesis.zip)
附件是我之前保存的cls文件，这个cls文件可以让文献处理正常，具体什么原因我也不知道。但是我没法再使用这个cls文件，因为有一些微调的格式在新的cls文件中，我也忘记了调整了啥。

**Comment by earthmoon** [comment]
我找到原因了，因为文献引用有这么两篇文献在其中：

> @Article{RN330,
author = {Tokatli, Cem and Mutlu, Ekrem and Arslan, Naime},
title = {Assessment of the potentially toxic element contamination in water of Şehriban Stream (Black Sea Region, Turkey) by using statistical and ecological indicators},
journal = {Water Environment Research},
volume = {93},
number = {10},
pages = {2060-2071},
ISSN = {1061-4303},
DOI = {10.1002/wer.1576},
url = {https://dx.doi.org/10.1002/wer.1576},
year = {2021},
type = {Journal Article}
}
@article{RN331,
   author = {Tokatli, Cem and Ugurluoglu, Alper and Kose, Esengul and Cicek, Arzu and Arslan, Naime and Dayioglu, Hayri and Emiroglu, Ozgur},
   title = {Ecological risk assessment of toxic metal contamination in a significant mining basin in Turkey},
   journal = {Environmental Earth Sciences},
   volume = {80},
   number = {1},
   note = {Times Cited: 2
0
2},
   abstract = {Emet River Basin is a worldwide significant mining area and contains the most important boron and chromium deposits of Turkey. In this study, water and sediment quality of Emet River Basin was evaluated using some toxic element risk assessment indices. Samples were collected seasonally from 8 locations and nickel (Ni), chromium (Cr), arsenic (As), zinc (Zn), lead (Pb), copper (Cu) and cadmium (Cd) accumulations in water and sediment were analysed using an ICP-OES. Water Quality Index (WQI), Heavy Metal Pollution Index (HPI), Heavy Metal Evaluation Index (HEI) and Degree of Contamination Index (C-deg) were applied to the data for assessing the water quality and Biological Risk Index (BRI), Potential Ecological Risk Index (PERI), Geo-Accumulation Index (I-geo), Contamination Factor (CF) and Pollution Load Index (PLI) were applied to the data for assessing the sediment quality. According to the results of WQI, HPI, HEI and C-deg, As and Cr were recorded as the most dangerous toxicants; according to the results of BRI, I-geo, CF and PLI, Cd, As and Pb were recorded as the most dangerous toxicants; and according to the results of PERI, Ni and Cr were recorded as the most dangerous toxicants among the investigated elements. The seasonal average values of applied ecological indices for the water of the basin ranged 71.41-888.29 for WQI, 85.97-915.76 for HPI, 4.04-49.80 for HEI and - 2.95-42.80 for C-deg. The seasonal average values of applied ecological indices for the sediment of the basin ranged 33.72-130.56 for PERI, 0.14-0.93 for BRI, - 3.92-5.99 for I-geo and 0.45-1.20 for PLI. In addition to the intensive agricultural applications around the region, the geological structure of the watershed was considered as the main cause of the high heavy metal accumulations in basin waters and sediments.},
   DOI = {10.1007/s12665-020-09333-4},
   url = {https://doi.org/10.1007/s12665-020-09333-4},
   year = {2021},
   type = {Journal Article}
}


引用后的状态为：
![image](https://user-images.githubusercontent.com/45897471/172972573-cf08d1be-7018-4b2f-9e20-7699acb7e3db.png)
像这种一作一样，又是同一年的，不应该以`2021a`和`2021b`区分吗？
